reconcile/utils/jira_client.py

```diff
@@ -22,20 +22,31 @@ def __init__(
         token_auth = self.secret_reader.read(token)
         self.jira = JIRA(self.server, token_auth=token_auth)
 
-    def get_issues(self, fields: Optional[Mapping] = None) -> GottenIssue:
+    def get_issues(self, fields: Optional[Mapping] = None) -> list[Issue]:
         block_size = 100
         block_num = 0
 
-        all_issues: GottenIssue = []
+        all_issues: list[Issue] = []
         jql = "project={}".format(self.project)
         kwargs: dict[str, Any] = {}
         if fields:
             kwargs["fields"] = ",".join(fields)
         while True:
             index = block_num * block_size
-            issues = self.jira.search_issues(jql, index, block_size, **kwargs)
-            # TODO: investigate why the arg-type ignore is needed in CI
-            all_issues.extend(issues)  # type: ignore[arg-type]
+            issues: Union[dict[str, Any], ResultList[Issue]] = self.jira.search_issues(
+                jql, index, block_size, **kwargs
+            )
+            if not isinstance(issues, ResultList):
+                # if search_issues was executed with json_result=True, then we have a Dict.
+                # However, we require a ResultList.
+                # See https://github.com/pycontribs/jira/commit/cc2508485c12232a4a9c4a56ee74175ed818ee20
+                logging.warning(
+                    "Jira client did not receive a ResultList."
+                    " Maybe the call was made with json_result=True which is"
+                    " currently not supported."
```

Comment:
It would be better to analyze `kwargs` for the unsupported `json_result` parameter and raise an exception. Then you can always expect `ResultList[Issue]` as the result type.

Reply:
My understanding is that we should not raise exceptions in the reconcile loop, because of the way reconcile counts the errors in some integrations. Since this is a utils class, I was trying to avoid bringing in exceptions in potentially unexpected places.

Reply:
Ah, ok, then the warning instead of the exception but expecting `ResultList[Issue]` as the result type is easier to understand.

Reply:
API documentation says it could be both. So +1 on checking the type this way.

Reply:
Adjusted `ResultList` -> `ResultList[Issue]`.

Reply:
Have a look here #2689. Depending on the type of the input it returns either a `list[dict]` or `list[RoleProtocol]` which is very similar here too. 
Defining the return type as `Union[Dict, List]` looks strange to me, therefore I would do something like

```python
while True:
    if kwargs.get("json_result"):
        logging.warning("...")
        continue
    issues: ResultList[Issue] = self.jira.search_issues(
                 jql, index, block_size, **kwargs
             )
    ...

`´`

Reply:
Further, Im not sure if mypy will be happy with the above suggestion. It has no knowledge of `json_result` -> it will expect a Union type as a return.